HyperV/scripts/utils.ps1

```diff
@@ -73,3 +73,80 @@ function GitClonePull($path, $url, $branch="master")
         }
     }
 }
+
+function Exec-PipInstall{
+    [CmdletBinding()]
+    param (
+        [parameter(Mandatory=$true)]
+        [string[]]$packages,
+
+        [parameter(Mandatory=$false)]
+        [switch]$usewheel,
+
+        [parameter(Mandatory=$false)]
+        [switch]$noindex,
+
+        [parameter(Mandatory=$false)]
+        [string]$findlinks=$null,
+
+        [parameter(Mandatory=$false)]
+        [switch]$Upgrade,
+
+        [parameter(Mandatory=$false)]
+        [switch]$pre
+    )
+    PROCESS {
+        foreach($package in $packages){
+            $command = "& pip install "
+            if ($Upgrade.IsPresent){
+                $command += "-U "
+            }
+            if ($pre.IsPresent){
+                $command += "--pre "
+            }
+            if ($usewheel.IsPresent){
+                $command += "--use-wheel "
+            }
+            if ($noindex.IsPresent){
+                $command += "--no-index "
+            }
+            if ($findlinks -ne $null -and $findlinks.Length -ne 0){
+                $command += "--findlinks=$findlinks "
+            }
+            $command +=  $package
+            ExecRetry {
+                $command
+                if ($LastExitCode) { 
+                    Throw "pip failed to install $package" 
+                }
+            }
+        } 
+    }
+}
+
+function Exec-EasyInstall{
+    [CmdletBinding()]
+    param (        
+        [parameter(Mandatory=$true)]
+        [string]$packages,
+
+        [parameter(Mandatory=$false)]
+        [string]$URL=$null        
+)
+    PROCESS {
+        foreach($package in $packages)
+        {   
+            $command = "& easy_install "
```

Comment:
Just a suggestion. This can be written as:

```
$command = @("easy_install")
```

Later you can add more options by simply doing:

```
$command += @("-f", $URL)
```

Then execute it like so:

```
& $command[0] $command[1..$command.Length]
```

I think you can even skip the & (test it to be sure).

This way you no longer need to worry about leaving a trailing space at the end of the command string.